vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -438,20 +397,27 @@ static float get_connection_box_lookahead_map_cost(int from_node_ind,
     ssize_t dx = ssize_t(from_canonical_loc->first) - ssize_t(box_location.first);
     ssize_t dy = ssize_t(from_canonical_loc->second) - ssize_t(box_location.second);
 
-    int from_seg_index = g_cost_map.node_to_segment(from_node_ind);
-    Cost_Entry cost_entry = g_cost_map.find_cost(from_seg_index, box_id, dx, dy);
+    int from_seg_index = cost_map_.node_to_segment(from_node_ind);
+    Cost_Entry cost_entry = cost_map_.find_cost(from_seg_index, box_id, dx, dy);
+
+    if (!cost_entry.valid()) {
+        // there is no route
+        return std::numeric_limits<float>::infinity();
+    }
+
     float expected_delay = cost_entry.delay;
     float expected_congestion = cost_entry.congestion;
 
     float expected_cost = criticality_fac * expected_delay + (1.0 - criticality_fac) * expected_congestion;
+    VTR_ASSERT(std::isfinite(expected_cost) && expected_cost >= 0.f);
     return expected_cost;
 }
 
 /* runs Dijkstra's algorithm from specified node until all nodes have been
  * visited. Each time a pin is visited, the delay/congestion information
  * to that pin is stored to an entry in the routing_cost_map */
 static void run_dijkstra(int start_node_ind,
-                         t_routing_cost_map* routing_cost_map) {
+                         routing_cost_list& routing_costs) {
```

Comment:
Please leave this as a pointer.  Mutating references is bad style.

Reply:
That is Google's style, but doesn't seem to be VtR's style.